packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -0,0 +1,112 @@
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import { Construct, Duration, Lazy, Stack } from '@aws-cdk/core';
+import { CfnAuthorizer } from '../../lib/apigateway.generated';
+import { CustomAuthorizer } from '../../lib/authorizer';
+
+/**
+ * Properties for TokenAuthorizer
+ */
+export interface TokenAuthorizerProps {
+
+  /**
+   * An optional name for the authorizer. When provided, this will also be used for the physical id of the
+   * CloudFormation resource of type `AWS::ApiGateway::Authorizer`.
+   *
+   * @default - CDK will use the uniqueId assigned to this construct.
+   */
+  readonly name?: string;
+
+  /**
+   * The authorizer lambda function.
+   */
+  readonly function: lambda.IFunction;
```

Comment:
We normally use `handler` to avoid using `function` since it's a keyword in JavaScript.

Reply:
I was expecting some information (or reference) here on how to implement this lambda? What is the protocol between APIGW and lambda?